### PR TITLE
add grammar support for modelChainConnection

### PR DESCRIPTION
**Pull request by MauricioUyaguari**
grammar and grammar compilation support for model chain connections

### CODE REVIEW

File: legend-engine-language-pure-grammar/src/main/antlr4/org/finos/legend/engine/language/pure/grammar/from/antlr4/connection/modelConnection/ModelConnectionLexerGrammar.g4 [hunk 1]
```diff
@@ -7,3 +7,4 @@ import CoreLexerGrammar;
 
 MODEL_CONNECTION_CLASS:                 'class';
 URL:                                    'url';
+MODEL_CHAIN_CONNECTION_MAPPINGS:        'mappings';
```
Comment (akphi) [inline legend-engine-language-pure-grammar/src/main/antlr4/org/finos/legend/engine/language/pure/grammar/from/antlr4/connection/modelConnection/ModelConnectionLexerGrammar.g4:10]: please add a new line at the end of file

File: legend-engine-language-pure-grammar/src/main/java/org/finos/legend/engine/language/pure/grammar/to/HelperConnectionGrammarComposer.java [hunk 2]
```diff
@@ -36,6 +37,9 @@ public static String getConnectionValueName(Connection connection, PureGrammarCo
         else if (connection instanceof XmlModelConnection)
         {
             return "XmlModelConnection";
+        } else if (connection instanceof ModelChainConnection)
```
Comment (akphi) [inline legend-engine-language-pure-grammar/src/main/java/org/finos/legend/engine/language/pure/grammar/to/HelperConnectionGrammarComposer.java:40]: formatting please

File: legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/store/modelToModel/connection/ModelChainConnection.java [hunk 3]
```diff
@@ -22,6 +23,7 @@
 public class ModelChainConnection extends Connection
 {
     public List<String> mappings;
+    public SourceInformation mappingsSourceInformation;
```
Comment (akphi) [inline legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/store/modelToModel/connection/ModelChainConnection.java:26]: it really upsets me that we still need to do this 😞 if only we could use mapping pointers

**Review by akphi — CHANGES_REQUESTED** [review]
@MauricioUyaguari I think the logic is fairly simple on this one. Just left some comments mostly on formatting 😭

File: legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/store/modelToModel/connection/ModelChainConnection.java (hunk 3 above)
Comment (MauricioUyaguari) [reply legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/store/modelToModel/connection/ModelChainConnection.java:26]: yup.

**Pull request merged**